src/NServiceBus.Core/Sagas/SagaPersistenceBehavior.cs

```diff
@@ -21,6 +21,19 @@ public SagaPersistenceBehavior(ISagaPersister persister, ICancelDeferredMessages
 
         public async Task Invoke(IInvokeHandlerContext context, Func<IInvokeHandlerContext, Task> next)
         {
+            var isTimeoutMessage = IsTimeoutMessage(context.Headers);
+            var isTimeoutHandler = context.MessageHandler.IsTimeoutHandler;
+
+            if (isTimeoutHandler && !isTimeoutMessage)
+            {
+                return;
+            }
+
+            if (!isTimeoutHandler && isTimeoutMessage)
+            {
+                return;
```

Comment:
is there a risk of loosing a timeout message in case you're requesting a timeout for a message and not implementing the IHandleTimeout interface?